### PR TITLE
fix: Server does not start via CLI when `auth` option is set

### PR DESCRIPTION
**Pull request by dblythy**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/parse-server/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/parse-server/blob/alpha/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

## Issue

Closes: https://github.com/parse-community/parse-server/issues/8665

## Approach

See Related: https://github.com/parse-community/parse-server/pull/8666.

### CODE REVIEW

**Comment by dblythy** [comment]
@FransGH please try this one out:

`npm install dblythy/parse-server#fix-auth-parser-lts`

**Comment by FransGH** [comment]
Just deployed `dblythy/parse-server#fix-auth-parser-lts` to staging and all seems to work fine!

Thanks Daniel for addressing this so quickly :)

**Review by FransGH — APPROVED** [review]

**Comment by dblythy** [comment]
Thanks for reviewing @FransGH!

**Comment by mtrezza** [comment]
@dblythy could you please add the fix to the PR description, and also a link to the alpha branch PR (if there is one)? Thanks!

**Comment by dblythy** [comment]
Sorry, the alpha PR is #8666

**Comment by FransGH** [comment]
@mtrezza not the most important but the problem is that the server fails to start via CLI when auth is configured (as opposed to just auth-adapters not loading)

**Comment by mtrezza** [comment]
Thanks for pointing this out, the PR title will be the changelog entry, so it's important we get this right. The server fails to start via CLI when auth is configured as a CLI parameter? If so, how about this title:

> fix: Server does not start via CLI with `--auth` parameter set

**Comment by FransGH** [comment]
I didn't try via --auth, only via a config.json. Assume however that it's independent of where the CLI gets the config from. Config file, cmd parameters and env-variables *should* have the same checks.

**Comment by mtrezza** [comment]
So you were using a config file and were passing that via the CLI?

**Comment by FransGH** [comment]
Yes, but only on staging and production. During dev a server instance is started via API where apparently the config is not so strictly checked.

**Comment by mtrezza** [comment]
Got it, different checks for CLI vs. API doesn't sound good. Do you want to open an issue to centralize the checks for config options regardless of how they are set? Sounds like a general issue.

How about this:

> fix: Server does not start via CLI when passing a config file with `auth` parameter set

**Comment by FransGH** [comment]
Thought about opening an issue for that but don't have time at the moment to contribute in fixing it. So not sure if it makes sense creating one?

**Comment by FransGH** [comment]
> fix: Server does not start via CLI when passing a config file with `auth` parameter set

"Fix: Server does not start via CLI when the auth option is set."

**Comment by mtrezza** [comment]
> Thought about opening an issue for that but don't have time at the moment to contribute in fixing it. So not sure if it makes sense creating one?

Sure, you don't need to contribute in fixing an issue - if you only want to report it that also helps.

**Comment by FransGH** [comment]
@mtrezza just for my understanding, what would be the approximate timeline for seeing this in a 5.x release update?
(I can work around this but would rather upgrade with an official release)

**Comment by mtrezza** [comment]
There seems to be an issue with Node 12 compatibility:

https://github.com/parse-community/parse-server/actions/runs/5386228844/jobs/9776051408?pr=8669#step:9:3989

File: spec/CLI.spec.js [hunk 1]
```diff
@@ -294,4 +294,20 @@ describe('execution', () => {
       done.fail(data.toString());
     });
   });
+
+  it('can start Parse Server with auth via CLI', done => {
+    const env = { ...process.env };
+    env.NODE_OPTIONS = '--dns-result-order=ipv4first';
```
Comment (mtrezza) [inline spec/CLI.spec.js]: There seems to be an issue with Node 12 compatibility:

> 1) execution can start Parse Server with auth via CLI
>   - Failed: node: --dns-result-order= is not allowed in NODE_OPTIONS

https://github.com/parse-community/parse-server/actions/runs/5386228844/jobs/9776051408?pr=8669#step:9:3989

It seems this needs a condition: Node >= 14

**Review by mtrezza — CHANGES_REQUESTED** [review]

File: spec/CLI.spec.js (hunk 1 above)
Comment (FransGH) [reply spec/CLI.spec.js]: @mtrezza I quickly patched the 5.x.x CLI.spec and tried to test with node 12 however node 12 is no longer support, see package.json:

```
  "engines": {
    "node": ">=14.21.0 <17 || >=18 <19"
  },

``` 

Assuming this is correct (which seems to make sense), the fix should be to remove testing on node 12?

**Review by FransGH — APPROVED** [review]

File: spec/CLI.spec.js (hunk 1 above)
Comment (dblythy) [reply spec/CLI.spec.js]: I resolved this because the LTS branch isn't tested on node 18, which requires the dns order. Tests should pass now

Comment (mtrezza) [reply spec/CLI.spec.js]: @FransGH Node 12 is still supported by Parse Server 5. You must have looked at a different branch.

**Review by mtrezza — COMMENTED** [review]
Looks good!

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.5.3](https://github.com/parse-community/parse-server/releases/tag/5.5.3)

**Pull request merged**